ros2topic/ros2topic/verb/echo.py

```diff
@@ -99,22 +105,34 @@ def main(args):
         if message_type is None:
             raise RuntimeError('Could not determine the type for the passed topic')
 
+        future = None
+        if args.once:
+            future = Future()
+            callback = subscriber_cb_once_decorator(callback, future)
+
         subscriber(
-            node, args.topic_name, message_type, callback, qos_profile)
+            node, args.topic_name, message_type, callback, qos_profile, future, args.timeout)
 
 
 def subscriber(
     node: Node,
     topic_name: str,
     message_type: MsgType,
     callback: Callable[[MsgType], Any],
-    qos_profile: QoSProfile
+    qos_profile: QoSProfile,
+    future = None,
+    timeout = None
 ) -> Optional[str]:
     """Initialize a node with a single subscription and spin."""
     node.create_subscription(
         message_type, topic_name, callback, qos_profile)
 
-    rclpy.spin(node)
+    if future == None:
+        rclpy.spin(node)
+    else:
+        rclpy.spin_until_future_complete(node, future, timeout)
+        if not future.done():
+            node.get_logger().error("Timeout occured after " + str(timeout) + " secconds: No message received on topic \"" + topic_name + "\"")
```

Comment:
@craigh92 consider using f-strings. Also, make sure your code passes the linter.

Reply:
I've removed the error message altogether. If `--timeout` can be used without `--once`, then timing out does not warrant an error message.